src/tools/ceph_objectstore_tool.cc

```diff
@@ -239,12 +232,9 @@ struct pgid_object_list {
         cout << std::endl;
       }
       f->open_array_section("pgid_object");
-      string pgid = i->first.c_str();
-      std::size_t pos = pgid.find("_");
-      if (pos == string::npos)
-        f->dump_string("pgid", pgid);
-      else
-        f->dump_string("pgid", pgid.substr(0, pos));
+      spg_t pgid;
+      if (i->first.is_pg(&pgid))
+        f->dump_string("pgid", stringify(pgid));
```

Comment:
I've reviewed just the objectstore tool changes.  They are fine as is.  I'll have to do something for the wip-cot-meta which can wait.